ext/soap/soap.c

```diff
@@ -525,9 +525,10 @@ PHP_METHOD(SoapParam, __construct)
 	if (zend_parse_parameters(ZEND_NUM_ARGS(), "zs", &data, &name, &name_length) == FAILURE) {
 		RETURN_THROWS();
 	}
+
 	if (name_length == 0) {
-		php_error_docref(NULL, E_WARNING, "Invalid parameter name");
-		return;
+		zend_argument_value_error(2, "cannot be empty");
```

Comment:
As far as I see, this PR will eliminate a few cases when a constructor returned null... :)